internal/pkg/template/templates/workloads/partials/cf/alb.yml

```diff
@@ -1,50 +1,51 @@
-TargetGroup:
+{{- range $i, $rule := .ALB.Rules}}
+TargetGroup{{ if ne $i 0 }}{{ $i }}{{ end }}:
   Metadata:
     'aws:copilot:description': "A target group to connect the load balancer to your service"
   Type: AWS::ElasticLoadBalancingV2::TargetGroup
   Properties:
-    HealthCheckPath: {{.HTTPHealthCheck.HealthCheckPath}} # Default is '/'.
-    {{- if .HTTPHealthCheck.Port}}
-    HealthCheckPort: {{.HTTPHealthCheck.Port}} # Default is 'traffic-port'.
+    HealthCheckPath: {{$rule.HTTPHealthCheck.HealthCheckPath}} # Default is '/'.
+    {{- if $rule.HTTPHealthCheck.Port}}
+    HealthCheckPort: {{$rule.HTTPHealthCheck.Port}} # Default is 'traffic-port'.
     {{- end}}
-    {{- if .HTTPHealthCheck.SuccessCodes}}
+    {{- if $rule.HTTPHealthCheck.SuccessCodes}}
     Matcher:
-      HttpCode: {{.HTTPHealthCheck.SuccessCodes}}
+      HttpCode: {{$rule.HTTPHealthCheck.SuccessCodes}}
     {{- end}}
-    {{- if .HTTPHealthCheck.HealthyThreshold}}
-    HealthyThresholdCount: {{.HTTPHealthCheck.HealthyThreshold}}
+    {{- if $rule.HTTPHealthCheck.HealthyThreshold}}
+    HealthyThresholdCount: {{$rule.HTTPHealthCheck.HealthyThreshold}}
     {{- end}}
-    {{- if .HTTPHealthCheck.UnhealthyThreshold}}
-    UnhealthyThresholdCount: {{.HTTPHealthCheck.UnhealthyThreshold}}
+    {{- if $rule.HTTPHealthCheck.UnhealthyThreshold}}
+    UnhealthyThresholdCount: {{$rule.HTTPHealthCheck.UnhealthyThreshold}}
     {{- end}}
-    {{- if .HTTPHealthCheck.Interval}}
-    HealthCheckIntervalSeconds: {{.HTTPHealthCheck.Interval}}
+    {{- if $rule.HTTPHealthCheck.Interval}}
+    HealthCheckIntervalSeconds: {{$rule.HTTPHealthCheck.Interval}}
     {{- end}}
-    {{- if .HTTPHealthCheck.Timeout}}
-    HealthCheckTimeoutSeconds: {{.HTTPHealthCheck.Timeout}}
+    {{- if $rule.HTTPHealthCheck.Timeout}}
+    HealthCheckTimeoutSeconds: {{$rule.HTTPHealthCheck.Timeout}}
     {{- end}}
-    {{- if .HealthCheckProtocol}}
-    HealthCheckProtocol: {{.HealthCheckProtocol}}
+    {{- if $rule.HealthCheckProtocol}}
+    HealthCheckProtocol: {{$rule.HealthCheckProtocol}}
     {{- end}}
-    Port: !Ref TargetPort
-    {{- if .HTTPTargetContainer.IsHTTPS }}
+    Port: {{$rule.TargetPort}}
+    {{- if eq $rule.TargetPort "443" }}
     Protocol: HTTPS
     {{- else }}
     Protocol: HTTP
     {{- end }}
-    {{- if .HTTPVersion}}
-    ProtocolVersion: {{.HTTPVersion}}
+    {{- if $rule.HTTPVersion}}
+    ProtocolVersion: {{$rule.HTTPVersion}}
     {{- end}}
     TargetGroupAttributes:
       - Key: deregistration_delay.timeout_seconds
-        Value: {{.DeregistrationDelay}} # ECS Default is 300; Copilot default is 60.
+        Value: {{$.DeregistrationDelay}} # ECS Default is 300; Copilot default is 60.
       - Key: stickiness.enabled
-        Value: !Ref Stickiness
+        Value: {{$rule.Stickiness}}
     TargetType: ip
     VpcId:
       Fn::ImportValue:
         !Sub "${AppName}-${EnvName}-VpcId"
-
+    {{- end}}
```

Comment:
nit: Does this end `{{- range $i, $rule := .ALB.Rules}}`? If so, can we fix the indentation so that they are aligned?

Reply:
+1 and we could add comment like 

```
{{- end }}{{/* range $topic := .Subscribe.Topics */}}
```